java/test/jmri/managers/JmriUserPreferencesManagerTest.java

```diff
@@ -750,33 +808,114 @@ public void testSetMultipleChoiceOption_3args_2() {
 
     @Test
     public void testGetClassDescription_0args() {
+        JmriUserPreferencesManager m = new JmriUserPreferencesManager();
+        Assert.assertEquals("Preference Manager", m.getClassDescription());
     }
 
     @Test
     public void testGetClassName() {
+        JmriUserPreferencesManager m = new JmriUserPreferencesManager();
+        Assert.assertEquals(JmriUserPreferencesManager.class.getName(), m.getClassName());
     }
 
     @Test
     public void testGetClassPreferences() {
+        JmriUserPreferencesManager m = new TestJmriUserPreferencesManager();
+        Assert.assertNull(m.getClassPreferences(strClass));
+        m.setPreferenceState(strClass, "test1", true);
+        Assert.assertNotNull(m.getClassPreferences(strClass));
+        Assert.assertEquals(1, m.getClassPreferences(strClass).getPreferenceListSize());
+        Assert.assertEquals("test1", m.getClassPreferences(strClass).getPreferenceList().get(0).getItem());
     }
 
     @Test
     public void testGetPreferencesSize() {
-    }
+        JmriUserPreferencesManager m = new TestJmriUserPreferencesManager();
+        Assert.assertNull(m.getClassPreferences(strClass));
+        Assert.assertEquals(0, m.getPreferencesSize(strClass));
+        m.setPreferenceState(strClass, "test1", true);
+        Assert.assertNotNull(m.getClassPreferences(strClass));
+        Assert.assertEquals(1, m.getPreferencesSize(strClass));
 
-    @Test
-    public void testReadUserPreferences() {
     }
 
     @Test
-    public void testSaveElement() {
+    @Ignore // identical to testSaveElement; I do not understand why it fails
```

Comment:
I don't see anything obvious.

I'll take another look at it tomorrow, when I have time to download the branch and test it locally.